workspaces/diff-engine/src/events/mod.rs

```diff
@@ -102,3 +106,33 @@ impl From<avro_rs::Error> for EventLoadingError {
     EventLoadingError::Avro(err)
   }
 }
+
+// EventContext
+// ------------
+
+#[derive(Deserialize, Debug, PartialEq, Serialize, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct EventContext {
+  client_id: String,
+  client_session_id: String,
+  client_command_batch_id: String,
+  created_at: String,
+}
+
+impl From<CommandContext> for EventContext {
+  fn from(command_context: CommandContext) -> Self {
+    Self {
+      client_id: command_context.client_id,
+      client_session_id: command_context.client_session_id,
+      client_command_batch_id: command_context.client_command_batch_id,
+      created_at: command_context
+        .created_at
+        .with_timezone(&Local)
+        .to_rfc3339(),
+    }
+  }
+}
+
+pub trait WithEventContext {
+  fn with_event_context(&mut self, event_context: EventContext);
+}
```

Comment:
Trait implemented by each event so that we can set their event context generically.